examples/plugins/project.json

```diff
@@ -72,5 +72,5 @@
       ]
     }
   },
-  "tags": ["scope:internal", "type:feature"]
+  "tags": ["scope:internal", "scope:core", "type:feature"]
```

Comment:
I had to add that to be able to use `executePlugin` from the `core` package in the e2e tests. This helps me to test the plugin logic without execution the cli.

Reply:
Otherwise it says:  
`4:1  error  A project tagged with "scope:core" can only depend on libs tagged with "scope:core", "scope:shared"  @nx/enforce-module-boundaries
`

Here graph:
<img width="733" alt="Screenshot 2024-02-28 at 15 14 30" src="https://github.com/code-pushup/cli/assets/10064416/c7563f76-80b1-48b8-9360-0177866337c4">

The reason:
I use `executePlugin` to e2e test the example lighthouse plugin in `e2e/examples-plugins-e2e`. This is the dependency that causes the issue.

Reply:
> Otherwise it says:
> ```4:1  error  A project tagged with "scope:core" can only depend on libs tagged with "scope:core", "scope:shared"  @nx/enforce-module-boundaries ```

What file is that?

Reply:
An E2E test only accesses the application via regular user input into the CLI.
Therefore, this resulting test cannot by definition be an E2E test (a user cannot just run `executePlugin` via our CLI) but an integration test of the plugin with `core`.

So instead of going down this rabbit hole, we should skip this test until such time that it can be converted into a valid E2E test that is compatible with our testing strategy. The tested logic is an example plugin, not a critical path through the system so it is not a big deal to test it once the logic is ready in an official plugin.

Reply:
@Tlacenka 
I had so far a different understanding of e2e... it tests the logic end to end. Like in real life.
My test did that. The plugin is executed in the same way it would be in real life. I can delete the test, or I could just add couple of seconds for this edge case... I am happy if we just delete it too.

@matejchalk 
> I use executePlugin to e2e test the example lighthouse plugin in e2e/examples-plugins-e2e. This is the dependency that causes the issue.

In `lighthouse.e2e.test` I use `executePlugin`.

Reply:
Update: 

I deleted the e2e tests. If we need them I'm happy to open a new PR

Reply:
> I had so far a different understanding of e2e... it tests the logic end to end. Like in real life.
> My test did that. The plugin is executed in the same way it would be in real life.

Using your logic, any test would be an E2E test, even just calling one function that would normally be a unit test because that is how the function would be called in real life (all tests should test important scenarios for a given unit of code).

An E2E test should test things start to finish as it would be in real life, yes. Our E2E tests are black-box, meaning they should **only use tools at real user's disposal**. A real user will only be able to **install and run the CLI package** and pass arguments to it. They will not run `executePlugin` or any `core` functions on demand (`collect` function in `core` is debatable but no further than that).

Reply:
As tests are removed and scopes back to unchanged i close this here